docs/manual/scala/guide/cluster/DomainModelling.md

```diff
@@ -1,19 +1,212 @@
+
+Each command has a `replyTo:ActorRef[R]` field where `R` is the reply that will be sent back to the caller. Replies are used to communicate back if a command was accepted or rejected or to read the aggregate data (ie: read-only commands). It's also possible to have a mix of both, for example, if the command succeeds it returns some updated data, if it fails it returns a rejected message. It's also possible to have commands without replies (ie: fire-and-forget). This is a less common pattern in CQRS Aggregate modelling though and not covered in this example.
+
+In Akka Typed, it's not possible to return an exception to the caller. All communication between the actor and the caller must be done via the `replyTo:ActorRef[R]` passed in the command. Therefore, if you want to signal a rejection, you most have it encoded in your reply protocol.
+
+```scala
+// Replies
+sealed trait Confirmation
+sealed trait Accepted extends Confirmation
+case object Accepted extends Accepted
+final case class Rejected(reason: String) extends Confirmation
+final case class CartState(items: Map[String, Int], status: String)
+
+// Commands
+sealed trait ShoppingCartCommand
+final case class UpdateItem(productId: String,
+                            quantity: Int,
+                            replyTo: ActorRef[Confirmation])
+  extends ShoppingCartCommand
+
+final case class Checkout(replyTo: ActorRef[Confirmation])
+  extends ShoppingCartCommand
+
+final case class Get(replyTo: ActorRef[CartState])
+```
+
+Note that we have different kinds of replies: `Confirmation` used when we want to modify the state. A modification request can be `Accepted` or `Rejected`. And `CartState` used when we want to read the state of the shopping cart. `CartState` is not the shopping cart itself, but the representation we want to expose to the external world.
+
+### Modelling Events
+
+Next we define the events that our model will be persisting. The events must extend Lagom's `AggregateEvent`. This is import for tagging the events. A topic we will cover a little further.
+
+```scala
+sealed trait ShoppingCartEvent extends AggregateEvent[ShoppingCartEvent] {
+  def aggregateTag = ShoppingCartEvent.Tag
+}
+
+final case class ItemUpdated(productId: String, quantity: Int)
+  extends ShoppingCartEvent
+
+final case object CheckedOut extends ShoppingCartEvent
+```
+
+### Defining Commands Handlers
+
+Once you define your protocol in terms of Commands, Replies, Events and State, you need to define the buiseness rules of your model. The command handlers define how to handle each incoming command, which validations must be applied and finally which events will be persisted, if any.
+
+You can encoding it in different ways. One popular style is to add the command handlers in your model classes. Since `ShoppingCart` have two state classes extensions, it makes sense to add the repective business rules validation on each state class. Each possible state will define how each command should be handled.
```

Comment:
I think that even more than "popular", this is the style guide recommended by Akka:

https://doc.akka.io/docs/akka/2.6/typed/persistence-style.html#event-handlers-in-the-state

It could be interesting to have a link to that page too.